lib/base/journaldlogger.cpp

```diff
@@ -0,0 +1,85 @@
+/* Icinga 2 | (c) 2021 Icinga GmbH | GPLv2+ */
+
+#if !defined(_WIN32) && defined(HAVE_SYSTEMD)
+#include "base/journaldlogger.hpp"
+#include "base/journaldlogger-ti.cpp"
+#include "base/configtype.hpp"
+#include "base/statsfunction.hpp"
+#include "base/sysloglogger.hpp"
+#ifndef SD_JOURNAL_SUPPRESS_LOCATION
+#define SD_JOURNAL_SUPPRESS_LOCATION
+#endif
+#include <systemd/sd-journal.h>
+
+using namespace icinga;
+
+REGISTER_TYPE(JournaldLogger);
+
+REGISTER_STATSFUNCTION(JournaldLogger, &JournaldLogger::StatsFunc);
+
+void JournaldLogger::StatsFunc(const Dictionary::Ptr& status, const Array::Ptr&)
+{
+	DictionaryData nodes;
+
+	for (const JournaldLogger::Ptr& journaldlogger : ConfigType::GetObjectsByType<JournaldLogger>()) {
+		nodes.emplace_back(journaldlogger->GetName(), 1); //add more stats
+	}
+
+	status->Set("journaldlogger", new Dictionary(std::move(nodes)));
+}
+
+void JournaldLogger::OnConfigLoaded()
+{
+	ObjectImpl<JournaldLogger>::OnConfigLoaded();
+
+	String facilityString = GetFacility();
+	m_Facility = SyslogFacilities::toNumber(facilityString);
+}
+
+void JournaldLogger::ValidateFacility(const Lazy<String>& lvalue, const ValidationUtils& utils)
+{
+	ObjectImpl<JournaldLogger>::ValidateFacility(lvalue, utils);
+	if (!SyslogFacilities::validate(lvalue()))
+		BOOST_THROW_EXCEPTION(ValidationError(this, { "facility" }, "Invalid facility specified."));
+}
+
+/**
+ * Processes a log entry and outputs it to journald.
+ *
+ * @param entry The log entry.
+ */
+void JournaldLogger::ProcessLogEntry(const LogEntry& entry)
+{
+	int severity;
+
+	switch (entry.Severity) {
+		case LogDebug:
+			severity = LOG_DEBUG;
+			break;
+		case LogNotice:
+			severity = LOG_NOTICE;
+			break;
+		case LogWarning:
+			severity = LOG_WARNING;
+			break;
+		case LogCritical:
+			severity = LOG_CRIT;
+			break;
+		case LogInformation:
+		default:
+			severity = LOG_INFO;
+			break;
+	}
```

Comment:
Given that this is duplicated from `SyslogLogger::ProcessEntry()`, I think it's worth putting it into a helper function like you already did for the facility stuff.